source/slang/glsl.meta.slang

```diff
@@ -2231,107 +2231,239 @@ __generic<T:__BuiltinFloatingPointType, let N:int>
 [require(glsl_hlsl_spirv, texture_sm_4_1_fragment)]
 public vector<T,4> textureProj(Sampler1D<vector<T,N>> sampler, vec2 p)
 {
-    return texture(sampler, p.x / p.y);
+    __target_switch
+    {
```

Comment:
```suggestion
    __requireComputeDerivative();
    __target_switch
    {
```

Since `textureProj` is not calling `texture()` which transitively calls `__requireComputeDerivative()` we must call `__requireComputeDerivative()` in this function to allow the implicit LOD function to work with compute shaders.

Reply:
I am trying to figure out how to prevent a regression of cases like this where `__requireComputeDerivative()` is missing.
Looking at the test you mentioned, `tests\glsl-intrinsic\compute-derivative\intrinsic-derivative-function-in-compute.slang`, I think it should be catching those cases.
But with my change, they are passing.
I am thinking we need a new issue for this case and improve the existing test, "intrinsic-derivative-function-in-compute.slang".
What do you think?

Reply:
The current `intrinsic-derivative-function-in-compute.slang` test is not done 'correctly'. To test it we would need 1 test per function.

Reply:
I found that the capability/extension is added by other functions and that might be a reason of why the existing test is passing.
I also found that we are adding more than just once and it ended up adding the same line many times.
```
OpCapability ComputeDerivativeGroupQuadsNV
OpCapability ComputeDerivativeGroupQuadsNV
OpCapability ComputeDerivativeGroupQuadsNV
OpCapability ComputeDerivativeGroupQuadsNV
OpCapability ComputeDerivativeGroupQuadsNV
OpCapability ComputeDerivativeGroupQuadsNV
OpCapability ComputeDerivativeGroupQuadsNV
OpCapability ComputeDerivativeGroupQuadsNV
OpCapability ComputeDerivativeGroupQuadsNV
OpCapability ComputeDerivativeGroupQuadsNV
OpCapability ComputeDerivativeGroupQuadsNV
OpCapability ComputeDerivativeGroupQuadsNV
OpCapability ComputeDerivativeGroupQuadsNV
OpCapability ComputeDerivativeGroupQuadsNV
OpCapability ComputeDerivativeGroupQuadsNV
OpCapability ComputeDerivativeGroupQuadsNV
OpCapability ComputeDerivativeGroupQuadsNV
OpCapability ComputeDerivativeGroupQuadsNV
```

Reply:
For the repeated OpCapability, I created a new issue, #4185

Reply:
I will create a new PR for the missing `__requireComputeDerivative()`.
I think it needs to be addressed with a new test and I want to separate it from this PR.

Reply:
> I found that the capability/extension is added by other functions and that might be a reason of why the existing test is passing.

yes, we need 1 test per function